sdk/eventgrid/azure-eventgrid/azure/eventgrid/_models.py

```diff
@@ -75,35 +76,53 @@ class CloudEvent(InternalCloudEvent, EventMixin):   #pylint:disable=too-many-ins
      unique for each distinct event.
     :type id: Optional[str]
     """
-
-    _validation = {
-        'source': {'required': True},
-        'type': {'required': True},
-    }
-
-    _attribute_map = {
-        'additional_properties': {'key': '', 'type': '{object}'},
-        'id': {'key': 'id', 'type': 'str'},
-        'source': {'key': 'source', 'type': 'str'},
-        'data': {'key': 'data', 'type': 'object'},
-        'data_base64': {'key': 'data_base64', 'type': 'bytearray'},
-        'type': {'key': 'type', 'type': 'str'},
-        'time': {'key': 'time', 'type': 'iso-8601'},
-        'specversion': {'key': 'specversion', 'type': 'str'},
-        'dataschema': {'key': 'dataschema', 'type': 'str'},
-        'datacontenttype': {'key': 'datacontenttype', 'type': 'str'},
-        'subject': {'key': 'subject', 'type': 'str'},
-    }
-
     def __init__(self, source, type, **kwargs):
         # type: (str, str, Any) -> None
-        kwargs.setdefault('id', uuid.uuid4())
-        kwargs.setdefault("source", source)
-        kwargs.setdefault("type", type)
-        kwargs.setdefault("time", dt.datetime.now(UTC()).isoformat())
-        kwargs.setdefault("specversion", "1.0")
-
-        super(CloudEvent, self).__init__(**kwargs)
+        self.source = source
+        self.type = type
+        self.specversion = kwargs.get("specversion", "1.0")
+        self.id = kwargs.get("id", str(uuid.uuid4()))
+        self.time = kwargs.get("time", dt.datetime.now(UTC()).isoformat())
+        self.data = kwargs.get("data", None)
+        self.datacontenttype = kwargs.get("datacontenttype", None)
+        self.dataschema = kwargs.get("dataschema", None)
+        self.subject = kwargs.get("subject", None)
+
+        for attr in kwargs:
+            setattr(self, attr, kwargs.get(attr))
+
+    @classmethod
+    def _from_generated(cls, generated, **kwargs):
+        if not generated:
+            return {}
```

Comment:
Why is that? from_generated will be used on models that comes from the server, I don't see how it could be `None`